pkg/applicationmanager/controllers/applicationrestore.go

```diff
@@ -679,10 +675,10 @@ func (a *ApplicationRestoreController) restoreVolumes(restore *storkapi.Applicat
 		// Portworx driver restore is not supported via job as it can be a secured px volume
 		// to access the token, we need to run the driver.startRestore in the same context as the stork controller
 		// this check is equivalent to (if !nfs || (nfs && driverName == volume.PortworxDriverName))
-		if !nfs || hasPXDdriver {
+		for driverName, vInfos := range backupVolumeInfoMappings {
 			// Here backupVolumeInfoMappings is framed based on driver name mapping, hence startRestore()
 			// gets called once per driver
-			for driverName, vInfos := range backupVolumeInfoMappings {
+			if !nfs || (nfs && driverName == volume.PortworxDriverName) {
```

Comment:
nit :: We can have a simplified check like
if !nfs || driverName == volume.PortworxDriverName {}

Reply:
Oops didnt realise the flag would regress like this.

Reply:
Yes, checking of driverName alone will working but I prefer check is gives clarity in understanding the logic :)